actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -231,6 +231,9 @@ class Parameters
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
     def initialize(parameters = {})
+      unless parameters.respond_to?(:with_indifferent_access)
```

Comment:
yes but imagine that, in case of a `NoMethodError` error in the method `with_indifferent_access` you catch this error and raise `ArgumentError`. Someone can override `with_indifferent_access` method and do some nasty stuff which can lead this error and in this case raising `ArgumentError` is much more confusing because even people will not be able to track the stack trace. I think we need to stick with your first implementation and check the responsibility of object with `respond_to?`.